Added known limitations documentation (#22)
CitrineInformatics/taurus-documentation: docs/known_limitations.md

# Known Limitations

## Purpose

> This set of documentation describes the next generation Citrine data format, code-named GEMD, that is independent of any implementation of the format.  

Reviewer: Isn't taurus the codename and GEMD the real name?

Author: Yes, and I don't know how my brain turned off there.

> This set of documentation describes the next generation Citrine data format, code-named GEMD, that is independent of any implementation of the format.  

Reviewer: ```suggestion
This set of documentation describes the next generation Citrine data format, code-named taurus, that is independent of any implementation of the format.  
```

*However*, as the Citrine platform is assumed to be the primary point of interaction between this data model and most users, known limitations of this model are documented here, with appropriate links to this page provided elsewhere in this documentation.

## Limitations

* The Citrine Platform cannot check for cycles in the material history.  

>     Material histories represent a chronology, so cycles in them represent time-travel (physically disallowed).  We do not have the ability to check that writing an object doesn't create a cycle in some material history.  

Reviewer: ```suggestion
    Material histories represent a chronology, so cycles in them represent time-travel (physically disallowed).
    We do not have the ability to check that writing an object doesn't create a cycle in some material history.  
```

* The Citrine Platform does not yet support Molecular Values (InChI, SMILES) or Molecular Bounds.  

* The Citrine Platform does not yet support Composition Values (Nominal Composition, Empirical Formula), although Composition Bounds are supported.   

> * The values of `name` and `labels` of Ingredient Runs are inherited from the associated Ingredient Specs, and thus are identically equal. However, this was not true in early designs.  This means some older implementations my still have those fields in Ingredient Runs and there are therefore order-of-operations concerns that might get in the way of object validation.  

Reviewer: ```suggestion
* The values of `name` and `labels` of Ingredient Runs are inherited from the associated Ingredient Specs, and thus are identically equal.
  However, this was not true in early designs.
  This means some older implementations my still have those fields in Ingredient Runs and there are therefore order-of-operations concerns that might get in the way of object validation.  
```

# Known Limitations

## Purpose

> This set of documentation describes the data format GEMD.  This data concept is independent of any implementation of the format.  *However*, as the Citrine platform is assumed to be the primary point of interaction between this data model and most users, known limitations of this model are documented here, with appropriate links to this page provided elsewhere in this documentation.

Reviewer: ```suggestion
This set of documentation specifies a data format, taurus, that is independent of any particular software implementation.
*However*, as the Citrine platform is likely to be the primary point of interaction between this data model and most users, known limitations of this model are documented here, with appropriate links to this page provided elsewhere in this documentation.
```

Reviewer: What's GEMD? :wink:

## Limitations

> * We cannot check for cycles in the material history.  

Reviewer: ```suggestion
* The Citrine Platform cannot check for cycles in the material history.  
```

    Material histories represent a chronology, so cycles in them represent time-travel (physically disallowed).  We do not have the ability to check that writing an object doesn't create a cycle in some material history.  

> * We have no support for Molecular Values (InChI, SMILES) or Molecular Bounds.  

Reviewer: ```suggestion
* The Citrine Platform does not yet support Molecular Values (InChI, SMILES) or Molecular Bounds.  
```

> * We have no support for Composition Values (Nominal Composition, Empirical Formula), although Composition Bounds are supported.   

Reviewer: ```suggestion
* The Citrine Platform does not yet support Composition Values (Nominal Composition, Empirical Formula), although Composition Bounds are supported.   
```

> * The values of name and labels fields of Ingredient Runs should be inherited from the associated Ingredient Specs, and thus be identically equal. However, due to implementation details, care must be taken around creating these objects:

Reviewer: I don't think we need this here, since the specification here no longer even has `name` and `labels` on IngredientRuns.  This is documentation that specifically belongs in `taurus` (the python binding), IMO.
